docs/upgrade.md

```diff
@@ -3,5 +3,263 @@ title: Upgrade guide
 description: Guide to update between major Powertools for AWS Lambda (TypeScript) versions
 ---
 
```

Comment:
Add a banner to signal a date or when they can expect to get an official date for v2. This will later be a section `End of support v1`

Reply:
Done!

I also added an announcement at the top of the site linking the upgrade guide 
<img width="555" alt="Screenshot 2024-02-09 at 16 53 54" src="https://github.com/aws-powertools/powertools-lambda-typescript/assets/7353869/9e0aabe9-36e6-40d0-a09b-8ad9868a39be">

I will also add a comment to the RFC once this is merged & published.